lib/pattern-preset.js

```diff
@@ -1,18 +1,27 @@
 const presets = {
 
-  'JavaScript': [
-    '.js',
-    '.jsx',
-    '.es6',
-  ],
+  'JavaScript': {
+    pathSubstrings: [
+      '.js',
+      '.jsx',
+      '.es6',
+    ],
+    githubClasses: [
+      'type-javascript',
+      'type-jsx',
+    ],
+  },
 
   'CoffeeScript': [
     '.coffee',
   ],
 
-  'TypeScript': [
-    '.ts',
-  ],
+  'TypeScript': {
+    pathSubstrings: ['.ts'],
+    githubClasses: [
+      'type-typescript',
+    ],
+  },
```

Comment:
Oops, not sure how I missed that one. I'll add it in, along with `type-coffeescript` above.